myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+            "nutritional_content[calcium]": f"{calcium}",
+            "nutritional_content[vitamin_c]": f"{vitamin_c}",
+            "nutritional_content[iron]": f"{iron}",
+            "food_entry[quantity]": "1.0",
+            "food_entry[meal_id]": "0",
+            "addtodiary": "no",
+            "preserve_exact_description_and_brand": "true",
+            "continue": "Save",
+        }
+        # Make entry public if requested, Hint: submit "sharefood": 0 also generates a public db entry, so only add
+        # "sharefood"" if really requested
+        if sharepublic:
+            data["sharefood"] = 1
+
+        url = parse.urljoin(self.BASE_URL_SECURE, SUBMIT_POST_PATH)
+        result = self.session.post(
+            url,
+            data,
+        )
+        if result.status_code == 200:
+            document = lxml.html.document_fromstring(result.content.decode("utf-8"))
+            try:
+                if not document.xpath(
+                    # If list is empty there should be no error, could be replaced with assert
+                    "//*[@id='errorExplanation']/ul/li"
+                ):
+                    return True
+                    # print("No Error :)")
+                    # Would like to return FoodItem, but seems that it take
+                    # to long until the submitted food is available in the DB
+                    # return self.get_food_search_results("{} {}".format(brand, description))[0]
+                else:  # Error occurred
+                    error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
+                    error = error.replace("Description ", "")  # For cosmetic reasons
+                    raise MyfitnesspalRequestFailed(
+                        f"Unable to submit food to MyFitnessPal: {error}"
+                    )
+            except Exception:
+                logger.warning(f"Unable to submit food to MyFitnessPal: {error}")
+
+        elif not result.ok:
+            logger.warning(
+                f"Request Error - Unable to submit food to MyFitnessPal: status code: {result.status_code}"
+            )
+            return None
+
+    def set_new_goal(
+        self,
+        energy: float = None,
+        energy_unit: str = None,
+        carbohydrates: float = None,
+        protein: float = None,
+        fat: float = None,
+        percent_carbohydrates: float = None,
+        percent_protein: float = None,
+        percent_fat: float = None,
+        # TODO Add micro nutritions
+        # saturated_fat: float = None,
+        # polyunsaturated_fat: float = None,
+        # monounsaturated_fat: float = None,
+        # trans_fat: float = None,
+        # fiber: float = None,
+        # sugar: float = None,
+        # cholesterol: float = None,
+        # sodium: float = None,
+        # potassium: float = None,
+        # vitamin_a: float = None,
+        # vitamin_c: float = None,
+        # calcium: float = None,
+        # iron: float = None,
```

Comment:
```suggestion
```

I'd really rather keep the code as clean as possible; so if you wouldn't mind, let's either remove these comments or actually implement them.